client/src/pages/people/Show.js

```diff
@@ -177,11 +177,13 @@ const PersonShow = ({ pageDispatchers }) => {
     (hasPosition && currentUser.isSuperUserForOrg(position.organization)) ||
     (person.role === Person.ROLE.PRINCIPAL && currentUser.isSuperUser())
   const canAddAssessment =
-    isAdmin ||
-    currentUser.position.associatedPositions
-      .filter(ap => ap.person)
-      .map(ap => ap.person.uuid)
-      .includes(person.uuid)
+    Position.isAdvisor(position) ||
+    (Position.isPrincipal(position) &&
+      (isAdmin ||
+        currentUser.position.associatedPositions
+          .filter(ap => ap.person)
+          .map(ap => ap.person.uuid)
+          .includes(person.uuid)))
```

Comment:
Maybe I misunderstand, but can we distinct the `allowed-to-add` rules and the `should-add`. One is about permissions, the other about checing the right conditions.

e.g.:

const shouldAddAssesment = Position.isPrincipal(position);

const allowedToAddAssesment = (isAdmin ||
        currentUser.position.associatedPositions
          .filter(ap => ap.person)
          .map(ap => ap.person.uuid)
          .includes(person.uuid)))

const canAddAssessment = shouldAddAssesment && allowedToAddAssesment;